microsoft-edge/progressive-web-apps-chromium/how-to/widgets.md

```diff
@@ -0,0 +1,495 @@
+    await self.registration.periodicSync.unregister(widget.definition.tag);
+  }
+}
+
+// Listen to periodicsync events to update all widget instances
+// periodically.
+self.addEventListener("periodicsync", async event => {
+  const widget = await self.widgets.getByTag(event.tag);
+
+  if (widget && "update" in widget.definition) {
+    event.waitUntil(updateWidget(widget));
+  }
+});
+
+async function updateWidget(widget) {
+  // Get the template and data URLs from the widget definition.
+  const templateUrl = widget.definition.msAcTemplate;
+  const dataUrl = widget.definition.data;
+
+  // Fetch the template text and data.
+  const template = await (await fetch(templateUrl)).text();
+  const data = await (await fetch(dataUrl)).text();
+
+  // Render the widget with the template and data.
+  await self.widgets.updateByTag(widget.definition.tag, {template, data});
+}
+```
+
+
+<!-- ====================================================================== -->
+## Demo app
+
+PWAmp is a music player PWA demo application that defines a widget. The PWAmp widget lets users visualize the current song and play the previous or next songs.
+
+1. Follow the instructions in [enable widgets in Microsoft Edge](#enable-widgets-in-microsoft-edge).
+1. Go to [PWAmp](https://microsoftedge.github.io/Demos/pwamp/) and install the app on Windows 11.
+1. Open the Windows 11 widgets board by pressing `Ctrl+W`.
+1. Click **Add widgets** to open the **widgets settings** screen, scroll to the **PWAmp mini player** widget and add it.
+1. Close the **widgets settings** screen. The **PWAmp mini player** is now displayed in the widgets board.
+
+The PWAmp widget displays the current song and buttons to play the previous or next song.
+
+![Windows widgets board, next to the PWAmp demo app. The widgets board contains the PWAmp mini player widget, showing the current song playing in the PWAmp app](../media/pwamp-widget.png)
+
+
+<!-- ====================================================================== -->
+<!-- widgets are too early in the making for MDN. But there is enough complexity with the APIs that a reference doc is needed.
+For now we are documenting all functions and objects in this section. When widgets become standard, move this to MDN and remove the section. -->
+## Service Worker API reference
+
+The service worker global object (or [ServiceWorkerGlobalScope](https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope)) contains a `widgets` attribute that exposes the following Promise-based methods:
+
+| Method | Description | Parameters | Return value |
+|:--- |:--- |:--- |:--- |
+| `getByTag(tag)` | Get a widget by tag | The widget tag | A Promise that resolves to the [widget object](#widget-object) that matches the tag, or `undefined`. |
+| `getByInstanceId(id)` | Get a widget by instance ID | The widget instance ID | A Promise that resolves to the corresponding [widget object](#widget-object), or undefined.
```

Comment:
```suggestion
| `getByInstanceId(id)` | Get a widget by instance ID | The widget instance ID | A Promise that resolves to the corresponding [widget object](#widget-object), or `undefined`.
```
GLOBAL: check "undefined" and "String" for needing backticks. If not a regular English sentence word, but a word used in code, might need backticks.